### DISCUSSION TITLE
Uploading Image Files Via Svelte

### DISCUSSION THREAD

**Discussion by samfcheng** [pocketbase/pocketbase#1907] [Q&A] [answered]
Hey,

I'm making a simple instagram-like photosharing site.
First, I bind the value of the input to a variable
![image](https://user-images.githubusercontent.com/51756184/221259954-335caf85-0f55-4240-b90f-4aa05c2b6b4e.png)
Then, submit it to pocketbase
![image](https://user-images.githubusercontent.com/51756184/221260100-1023b408-5ce0-4e82-b73b-6ea18a635655.png)
Here is the collection in pocketbase with some dummy data:
![image](https://user-images.githubusercontent.com/51756184/221260237-6822d036-9d65-47bf-9816-97dcd3e83d55.png)
And here's the error details:
![image](https://user-images.githubusercontent.com/51756184/221260347-4a93f2f7-8871-4207-be2a-b982757eb8b5.png)

How can I make this work?

**Comment by ganigeorgiev** [comment, marked as answer]
From the documentation in https://svelte.dev/docs#template-syntax-element-directives-bind-property:

> On <input> elements with type="file", you can use **bind:files** to get the [FileList of selected files](https://developer.mozilla.org/en-US/docs/Web/API/FileList). It is readonly.

Additionally, I don't think you can pass directly a FileList as `FormData` field value. From the [`FormData` documentation](https://developer.mozilla.org/en-US/docs/Web/API/FormData/set):

> The field's value: **This can be a string or Blob** (including subclasses such as File). If none of these are specified the value is converted to a string.

Or in other words:

1. Change `bind:value` to `bind:files`.
2. Iterate the result `FilesList` and `append` each individual `File` instance.

**Reply by samfcheng** [reply]
I tried that originally, it doesn't work
![image](https://user-images.githubusercontent.com/51756184/221298814-a66e67fe-4b17-44b0-8d54-d12fc975f6b4.png)
![image](https://user-images.githubusercontent.com/51756184/221298845-4458af75-ef25-4f66-9481-88043ae99004.png)
I did that following the instruction on the [pocketbase docs](https://pocketbase.io/docs/files-handling).
![image](https://user-images.githubusercontent.com/51756184/221299015-b3a0485c-361c-4f81-a65e-6b7524400671.png)
It seems like maybe since there's only one file that's uploaded it doesn't work?
Just `fileInput` doesn't work either (Invalid formatting again)
![image](https://user-images.githubusercontent.com/51756184/221299298-f2f2013e-fa62-41be-8f89-011408519a51.png)

**Reply by ganigeorgiev** [reply]
@Samalander0 Please see 2) of my previous comment.

You need to pass a `File` (or `Blob`) instance. In your example you are passing the entire `FileList` not an item of it. The [PocketBase docs](https://pocketbase.io/docs/files-handling) uses plain js and `fileInput` there is the DOM element, not `FileList`.

`bind:files` returns a `FileList` and it will be serialized as string when set as `FormData` value. Instead, you have to either iterate over the `fileInput` items and append each individually OR append only the first file, aka.:

```js
formData.append("file", fileInput[0])

// OR iterate over the FileList entries

for (let file of fileInput) {
    formData.append("file", file);
}
```
_(I recommend to rename your `fileInput` variable to `fileList` or similar to avoid the confusion)_
